debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -168,22 +168,37 @@ public Lsn getMaxLsn() throws SQLException {
         }, "Maximum LSN query must return exactly one value"));
     }
 
-    public MaxLsnResult getMaxLsnResult(boolean skipLowActivityLsnsEnabled) throws SQLException {
-        if (skipLowActivityLsnsEnabled) {
-            return prepareQueryAndMap(GET_MAX_LSN_SKIP_LOW_ACTIVTY, statement -> {
-            }, singleResultMapper(rs -> {
-                final MaxLsnResult ret = new MaxLsnResult(Lsn.valueOf(rs.getBytes(1)), Lsn.valueOf(rs.getBytes(2)));
-                LOGGER.trace("Current maximum transactional LSN is {}", ret);
-                return ret;
-            }, "Maximum transactional LSN query must return exactly one value"));
-        }
-        Lsn maxLsn = prepareQueryAndMap(GET_MAX_LSN, statement -> {
+    /**
+     * @return the log sequence number of the most recent transaction
+     *         that isn't further than {@code maxOffset} from {@code lastLsn}.
+     */
+    public Lsn getNthTransactionLsn(Lsn lastLsn, int maxOffset) throws SQLException {
+        String replacement = lastLsn.isAvailable() ? " AND start_lsn >= ?" : "";
```

Comment:
Could we have two statements here instead of constructing it dynamically so the statement is chosen and paramas set depening on last LSN availability? I think it would make the code more maintainable

Reply:
Done.